intelmq/bots/collectors/xmpp/collector.py

```diff
@@ -21,32 +40,77 @@ def init(self):
             self.stop()
 
     def process(self):
-
         if self.xmpp is None:
-            self.xmpp = sleekxmpp.XMPPBot(self.parameters.xmpp_user + '@' + self.parameters.xmpp_server,
-                                self.parameters.xmpp_password,
-                                self.send_message,
-                                self.logger)
+            self.xmpp = XMPPClientBot(self.parameters.xmpp_user + '@'
+                                      + self.parameters.xmpp_server,
+                                      self.parameters.xmpp_password,
+                                      self.parameters.xmpp_room,
+                                      self.parameters.xmpp_room_nick,
+                                      self.parameters.xmpp_room_password,
+                                      self.logger)
             self.xmpp.connect(reattempt=True)
             self.xmpp.process()
 
+            # Add Handlers and register Plugins
+            self.xmpp.register_plugin('xep_0030')  # Service Discovery
+            self.xmpp.register_plugin('xep_0045')  # Multi-User Chat
+            self.xmpp.add_event_handler("message", self.log_message)
+            self.xmpp.add_event_handler("groupchat_message", self.log_message)
+
     def killbot(self):
         self.xmpp.disconnect(wait=True)
         self.logger.info("Disconnected")
 
+    def log_message(self, msg):
+        self.logger.debug("XMPP Received Event: %r , from %r", msg['body'],
+                          msg['from'])
+        self.logger.info("XMPP Event received")
 
-class XMPPOutputBot(sleekxmpp.ClientXMPP):
+        event = None
+        raw_msg = None
 
-    def __init__(self, jid, password, send_message, logger):
+        # Try to decode the message as json, it might be an event.
+        # if this is not possible treat it as raw.
+        # TODO: There might be additional work to decode the message
+        # as base64
+        try:
+            event = json.loads(msg['body'])
+        except:
+            self.logger.info("XMPP Could not interpret the message as json. "
+                             "Treating as Raw")
+            self.logger.debug("XMPP I'm not able to tell if it is"
```

Comment:
Please remove the `XMPP` at the beginning of all log outputs. The bot name will be inserted by another layer. Currently this is the outcome (without date etc):
> xmpp-collector: XMPP I'm not able to tell if it isbase64 or not!

And as you can see, there's a whitespace missing